### PR TITLE
cntlr can be none

### PR DESCRIPTION
**Pull request by austinmatherne-wk**
#### Reason for change
Remove not none assertions on cntlr field which can in fact be none

#### Description of change
Remove non none assertions

#### Steps to Test
* CI/mypy

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by sagesmith-wf — COMMENTED** [review]
What code path does this happen in? It seems odd?

File: arelle/FileSource.py [hunk 1]
```diff
@@ -200,7 +200,6 @@ def open(self, reloadCache: bool = False) -> None:
                 return  # an error should have been logged
             if self.isZip:
                 try:
-                    assert self.cntlr is not None
```
Comment (austinmatherne-wk) [inline arelle/FileSource.py:203]: arbitrary line for discussion:

> What code path does this happen in? It seems odd?

From building a [xule rule set](https://github.com/xbrlus/xule/blob/60354a037977f0e0442f0c92b265fc594501b99c/plugin/xule/XuleRuleSet.py#L159).

**Review by sagesmith-wf — APPROVED** [review]

**Review by aaroncameron-wk — APPROVED** [review]

File: arelle/FileSource.py (hunk 1 above)
Comment (sagesmith-wf) [reply arelle/FileSource.py:203]: Huh, is that the expected usage there? I haven't ever seen someone loading using the filesource directly

Comment (derekgengenbacher-wf) [reply arelle/FileSource.py:203]: Is this path hit when loading the XULE validation zips?

Comment (austinmatherne-wk) [reply arelle/FileSource.py:203]: > Huh, is that the expected usage there? I haven't ever seen someone loading using the filesource directly

Yes, the file stream methods specifically check for and handle the none cntlr case.

**Review by derekgengenbacher-wf — APPROVED** [review]

**Comment by derekgengenbacher-wf** [comment]
QA +10 Testing locally succeeded.

**Pull request merged**